src/main/java/core/basesyntax/service/impl/TransitionProcessorImpl.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.TransitionProcessor;
+import java.util.Map;
+
+public class TransitionProcessorImpl implements TransitionProcessor {
+    @Override
+    public void processTransaction(Map<String, Integer> balanceMap, FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        switch (transaction.getOperation()) {
+            case BALANCE -> balanceMap.put(fruit, quantity);
+            case SUPPLY, RETURN -> balanceMap.put(fruit,
+                        balanceMap.getOrDefault(fruit, 0) + quantity);
+            case PURCHASE -> balanceMap.put(fruit,
+                        balanceMap.getOrDefault(fruit, 0) - quantity);
+            default -> throw new IllegalArgumentException("Invalid transaction type: "
+                        + transaction.getOperation());
+        }
+
+        if (balanceMap.get(fruit) < 0) {
+            balanceMap.put(fruit, 0);
+        }
```

Comment:
Just adding 0 when there is an error in processing transaction is not a good approach, you should throw an exception for example.